project-koku/koku
# COST-953: GCP processing window fix

The GCP downloader `_generate_monthly_pseudo_manifest` function was generating a scan window that was one day off.  Additionally there was an assumption that the partitioning date is the same as the usage date.  This did not turn out to be the case and we are now doing the BigQuery query based on usage_start so that it aligns with the summary processor

**Testing**
1.  Set DATE_OVERRIDE to Jan-29 and ingest GCP.  Verify data comes in for 1-29 of January
2.  Set DATE_OVERRIDE to Jan-30 and ingest GCP.  Verify data comes in for 1-30 of January.  Verify 1-29 does not change.
3.  Set DATE_OVERRIDE to Jan-31 and ingest GCP.  Verify data comes in for 1-31 of January.  Verify 1-30 does not change.
4.  Set DATE_OVERRIDE to Feb 1 and ingest GCP.  Verify data comes in for 1 January - 1 Feb.  Verify January data does not change.

**Test Results**
[cost_593_ut.txt](https://github.com/project-koku/koku/files/5907703/cost_593_ut.txt)
